### ISSUE TITLE
RESTEasy Reactive client and server: make it trivial to dump HTTP traffic

### ISSUE THREAD

**Issue by FroMage** [quarkusio/quarkus#22213]
### Description

This mostly happens to me on the client, but I think it can also be useful for the server.

When using MP REST Client via RR Client, I'm doing a call via a proxy: `someclient.someMethod()` so I don't have access to the request/response and if something goes wrong and a header is missing, or an exception is thrown or whatever, it's not trivial at all to know why, because all this is hidden from the user.

I turned out to have an exceptional status code returned by my github client due to a missing header and I had to use a debugger or TCP inspector to see it, because RR was hiding that HTTP traffic from me.

I think we should add an option to dump all HTTP traffic with a property:

```properties
# Client and Server
quarkus.resteasy-reactive.dump-traffic=true
# Only server
quarkus.resteasy-reactive.server.dump-traffic=true
# Only client
quarkus.resteasy-reactive.client.dump-traffic=true
```

This would have saved me quite a few times.

**Comment by geoand** [comment]
> Perhaps it's not RR that should handle this but Vert.x, this way it works across the entire framework?

+1 on this - cc @cescoffier

**Comment by cescoffier** [comment]
Yes, for the server at least we can do that at the vertx level (there is an existing route handler) For the client, it's a bit more complicated, but I agree that we should provide that feature. 

How many times I got a 400 - Bad request without understanding what was passed and received?

**Comment by geoand** [comment]
I wonder if we could then use the dumps somehow in testing or code
generation...

**Comment by michalszynkiewicz** [comment]
@FroMage does https://github.com/quarkusio/quarkus/pull/21286 go in this direction?

**Comment by FroMage** [comment]
It does, but we could not find it, because it was not documented :(

So we need to document it.

Now I'm thinking we should have something in the DEV UI to make it easier to display.

**Comment by michalszynkiewicz** [comment]
it wouldn't be that hard to add sth to dev ui, we can easily collect requests and responses and we know how to do websockets with dev ui...

I added a few words to the docs:
https://github.com/quarkusio/quarkus/pull/22445

**Comment by FroMage** [comment]
Ideally we could even solve this at the vert.x level, because OIDC uses the Vert.x HTTP client and I also want to dump this.

Either logging on the console, or one file per request, but something as useful as you get from the browser network console.

**Comment by geoand** [comment]
If I get around to introducing pcap dump support into Vertx, then this would be achieved easily

**Comment by FroMage** [comment]
pcap would be nice, but a text file generated per request would be much simpler to use: `target/logs/{datetime}-{server/host-if-client}-{method}-{uri}.log` containing the dumped request/response with headers, like dev console does in browsers.
I mean, both solutions would be great:

```
quarkus.http.log=pcap
# for text files
quarkus.http.log=true
```

**Comment by geoand** [comment]
I came up with a patch in Vert.x (I also have something in Quarkus) to allow us to capture Vert.x based HTTP Server and Client traffic in a pcap file that can be opened with Wireshark and looks something like this:
![Screenshot from 2022-01-28 16-51-38](https://user-images.githubusercontent.com/4374975/151568244-34754ab1-5742-4c88-b1ce-2d63c0e48d63.png)

**Comment by kostacasa** [comment]
It's likely we'll want to enable trace logging in serverless environments that don't have a local filesystem. An environment variable that can instruct the server to dump the wire logs to console would be much more developer friendly. This is now possible for the client with the introduction of https://github.com/quarkusio/quarkus/pull/21286 and it has proven to be useful on multiple occasions. We would really prefer something similar for the server portion.

**Comment by geoand** [comment]
Vert.x will provide something similar hopefully in 4.3
